File: components/src/commonMain/kotlin/com/zero/components/core/ButtonComponent.kt
package com.zero.components.core

import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import com.arkivanov.decompose.extensions.compose.jetbrains.subscribeAsState
import com.zero.common.component.Component
import com.zero.models.core.ButtonModel

@Composable
fun ButtonComponent(component: Component) {
    val modelState = component.state.subscribeAsState()
    val model = modelState.value.componentModel as? ButtonModel

    model?.let {
        Button(onClick = model.onClick) {
            Text(model.text)
        }
    }
}

File: ios/src/iosAppMain/apple/composers/core/ButtonComponent.swift
//
// Created by Gabriel Timotei on 2/21/22.
//

import SwiftUI
import common

public struct ButtonComponent: View {
    @ObservedObject
    private var componentState: ObservableValue<ComponentState>

    public init(component: Component) {
        componentState = ObservableValue<ComponentState>(component.state)
    }

    public var body : some View {
        if let buttonModel = componentState.value.componentModel as? ButtonModel {
            Button(action: buttonModel.onClick) {
                Text(buttonModel.text)
            }
        } else {
            EmptyView()
        }
    }
}
